### PR TITLE
OS-supported feature based dependencies and RIOT

### PR DESCRIPTION
**Pull request by ibriano**
Introducing a novel way to mark the dependencies of a feature in our build
system, so we don't need to use PLATFORM_* to know if something is or not
supported. Instead, when kbuild starts, it will detect which OS is building
for based on that the dependency resolver says, and choose an OS-specific
file to include that selects the features we have support for that OS.

Then come some changes necessary to make Soletta build for upstream RIOT.

### CODE REVIEW

File: src/lib/comms/Makefile [hunk 1]
```diff
@@ -1,49 +1,38 @@
...
+    sol-network-riot.o
 
-    ifeq (y,$(SOL_PLATFORM_LINUX))
-    obj-comms-$(COMMS) += \
-        sol-network-linux.o
-    endif
-endif
+obj-networking-$(PLATFORM_LINUX) += \
+    sol-network-linux.o \
+    sol-socket-linux.o
```
Comment (barbieri) [inline src/lib/comms/Makefile]: please take the time and make these files follow the other implementation pattern "sol-XXXX-impl-linux".

File: src/lib/io/Kconfig [hunk 2]
```diff
@@ -1,3 +1,54 @@
...
+config USE_I2C
+    bool "I2C Support"
+    depends on FEATURE_HW_I2C
+    default y
+
+config USE_GPIO
+    bool "GPIO Support"
+    depends on FEATURE_HW_GPIO
+    default y
+
```
Comment (barbieri) [inline src/lib/io/Kconfig]: Please take the time to document all the hw access, try wikipedia's short description or something that can clarify to users that may not know what it is.

File: src/lib/io/Makefile [hunk 3]
```diff
@@ -1,122 +1,70 @@
...
-ifeq (y,$(PLATFORM_RIOTOS))
-ifeq (y, $(USE_AIO))
-obj-io-$(IO) += \
+obj-io-aio-$(USE_AIO) := \
+    sol-aio-common.o
+obj-io-aio-$(PLATFORM_RIOTOS) += \
     sol-aio-riot.o
-endif # USE_AIO
+obj-io-aio-$(PLATFORM_LINUX) += \
+    sol-aio-linux.o
```
Comment (barbieri) [inline src/lib/io/Makefile]: as in the other comment, take the time to rename to "-impl"

**Comment by ibriano** [comment]
I'll address those comments separately. Merged.

**Pull request closed without merging**